sdks/python/apache_beam/coders/row_coder.py

```diff
@@ -150,7 +150,9 @@ def _coder_from_type(field_type):
 def _nonnull_coder_from_type(field_type):
   type_info = field_type.WhichOneof("type_info")
   if type_info == "atomic_type":
-    if field_type.atomic_type in (schema_pb2.INT32, schema_pb2.INT64):
+    if field_type.atomic_type in (schema_pb2.INT16,
```

Comment:
From #19815 INT16 was encoded in BigEndianShortCoder in Java side but just not implemented in other SDKs. This may cause data corruption. The "Encountered a type that is not currently supported by RowCoder" Exception is currently correctly raised.